lib/class-wp-rest-widget-types-controller.php

```diff
@@ -0,0 +1,456 @@
+	}
+
+	/**
+	 * Retrieves the widget type' schema, conforming to JSON Schema.
+	 *
+	 * @since x.x.0
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		if ( $this->schema ) {
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'widget-type',
+			'type'       => 'object',
+			'properties' => array(
+				'name'            => array(
+					'description' => __( 'Unique name identifying the widget type.', 'gutenberg' ),
+					'type'        => 'string',
+					'default'     => '',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'id_base'         => array(
+					'description' => __( 'Unique name identifying the widget type.', 'gutenberg' ),
+					'type'        => 'string',
+					'default'     => '',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'option_name'     => array(
+					'description' => __( 'Option name.', 'gutenberg' ),
+					'type'        => 'string',
+					'default'     => '',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'widget_class'    => array(
+					'description' => __( 'Widget class name.', 'gutenberg' ),
+					'type'        => 'string',
+					'default'     => '',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'control_options' => array(
+					'description'          => __( 'Control options', 'gutenberg' ),
+					'type'                 => 'Object',
+					'default'              => array(),
+					'properties'           => array(),
+					'context'              => array( 'embed', 'view', 'edit' ),
+					'readonly'             => true,
+					'additionalProperties' => true,
+				),
+				'widget_options'  => array(
+					'description'          => __( 'Widget options', 'gutenberg' ),
+					'type'                 => 'Object',
+					'default'              => array(),
+					'properties'           => array(
+						'classname'                   => array(
+							'description' => __( 'Class name', 'gutenberg' ),
+							'type'        => 'string',
+							'default'     => '',
+							'context'     => array( 'embed', 'view', 'edit' ),
+							'readonly'    => true,
+						),
+						'customize_selective_refresh' => array(
+							'description' => __( 'Customize selective refresh.', 'gutenberg' ),
+							'type'        => 'boolean',
+							'default'     => false,
+							'context'     => array( 'embed', 'view', 'edit' ),
+							'readonly'    => true,
+						),
+						'description'                 => array(
+							'description' => __( 'Description of the widget.', 'gutenberg' ),
+							'type'        => 'string',
+							'default'     => '',
+							'context'     => array( 'view', 'edit', 'embed' ),
+						),
+					),
+					'additionalProperties' => array(
+						'type' => 'Object',
```

Comment:
```suggestion
						'type' => 'object',
```

Also, are we sure this is accurate? Why can't additional args be of any type?

Reply:
Fixed https://github.com/WordPress/gutenberg/pull/26042/commits/7d3884e82dbe50581da253ebb95580046e096d20